src/components/autocomplete/test/e2e/autocomplete.js

```diff
@@ -0,0 +1,125 @@
+import { VlElement, By } from '../../../../utils/test';
+
+export class VlTestAutocomplete extends VlElement {
+
+  async getPlaceHolder() {
+    const input = await this.getElementInShadow(this, 'input');
+    return input.getAttribute('placeholder');
+  }
+
+  async isDisabled() {
+    return this.hasAttribute('disabled');
+  }
+
+  async setInputValue(value)
+  {
+    const input = await this.getElementInShadow(this, 'input');
+    input.sendKeys(value);
+  }
+
+  async assertSuggestionsCount(count)
+  {
+    await this.waitUntilShadowDomElementsCount(this, By.css(`ul[class='vl-autocomplete__list'] > li[class*='uig-autocomplete-item']`), count)
+  }
+
+  async getSuggestions()
+  {
+    await this.waitUntilShadowDomElementLocated(this, By.css('div[class=vl-autocomplete]:not([hidden=true])'))
+
+    const list = await this.findShadowDomElements(this, By.css(`ul[class='vl-autocomplete__list'] > li[class*='uig-autocomplete-item']`));
+
+    const asyncItemInfos = [];
+    list.forEach( e => {
+      asyncItemInfos.push(this.getItemInfo(e));
+    })
+
+    return Promise.all(asyncItemInfos).then(items => items);
+  }
+
+  async getGroupInfos()
+  {
+    await this.waitUntilShadowDomElementLocated(this, By.css('div[class=vl-autocomplete]:not([hidden=true])'))
+    const list = await this.findShadowDomElements(this, By.css(`ul[class='vl-autocomplete__list'] > li[class*='uig-autocomplete-group']`));
+    return Promise.all(list.map(this.getGroupInfo)).then(g => g);
+  }
+
+  async getGroupIndex(groupName)
+  {
+    const groups = await this.getGroupInfos();
+    const group = groups.find(g => g.name === groupName);
+    return group.index;
+  }
+
+  async getGroupInfo(e)
+  {
+    const name = await e.getText();
+    const index = await e.getAttribute("groupindex");
+
+    return ({name, index})
+  }
+
+  async getGroupNames()
+  {
+    const groups = await this.getGroupInfos();
+    return groups.map(g => g.name);
+  }
+
+  async getSuggestionsOfGroup(groupName)
+  {
+    const groupIndex = await this.getGroupIndex(groupName);
+
+    await this.waitUntilShadowDomElementLocated(this, By.css(`div[class='vl-autocomplete']:not([hidden=true])`))
+    const list = await this.findShadowDomElements(this, By.css(`ul[class='vl-autocomplete__list'] > li[class*='uig-autocomplete-item'][groupindex='${groupIndex}']`));
+
+    const asyncItemInfos = [];
+    list.forEach( e => {
+      asyncItemInfos.push(this.getItemInfo(e));
+    })
+
+     return Promise.all(asyncItemInfos).then(items => items);
+  }
+
+  async getItemTitleInfo(e)
+  {
+    const title = await e.getText();
+
+    return ({title})
+  }
+
+  async getItemInfo(item)
+  {
+    const itemInfo = {};
+
+    try {
+      itemInfo.title = await this.getItemProperty(item, "uig-autocomplete_title");
+    }
+    catch(e)
+    {
+      // title is optional, do nothing if item cannot be found
```

Comment:
Lege catch clauses?

Reply:
Idd die is leeg omdat die attributen optioneel zijn, als die niet gevonden kunnen worden mag de code niet falen maar moet die dus gewoon verder gaan.